src/lib/build.js

```diff
@@ -1,14 +1,34 @@
 const build = require('@netlify/build')
 
+const { getSiteInformation } = require('../utils/dev')
+
+const getBuildEnv = async ({ context }) => {
+  const { warn, error, netlify } = context
+  const { site, api } = netlify
+  const { teamEnv, addonsEnv, siteEnv, dotFilesEnv } = await getSiteInformation({
```

Comment:
We should reach a conclusion in https://github.com/netlify/build/issues/1514 before adding `.env` support in local builds. Would it be ok to split this specific feature from this PR?

Reply:
Done in https://github.com/netlify/cli/pull/1519/commits/1556c5694df980e1b53ffe8eb4fb2592d444ec76